modules/ui/stickyNotes.js

```diff
@@ -1,10 +1,24 @@
-const version = '2.0.1';
+const version = '2.1.0';
 
 let newStickyKeybindFunction;
 
 function injMod() {
+
+    goosemodScope.stickyNotes = {
```

Comment:
Do not pollute the global scope like this, why was this changed / used? Just use variables in the module.

Reply:
I more or less just followed the Visual Tweaks module as an example of how to make settings work. I also want them to save and it looked like Visual Tweaks knew how to do that.

Reply:
And for the reason I want it to save, there's no point setting the setting if it won't save across loads, because it's for something that happens on module load.

Reply:
Visual tweaks is very old so probably shouldn't be followed at the moment. I'd recommend just using variables instead of modifying the GooseMod scope.

Reply:
> And for the reason I want it to save, there's no point setting the setting if it won't save across loads, because it's for something that happens on module load.

What? With save settings and load settings you can just set a variable as the setting like in other modules.

Reply:
Ah, well idk how much of the module things work because theres very little documentation on it, but good to know.

Reply:
> Ah, well idk how much of the module things work because theres very little documentation on it, but good to know.

Fair enough.

Reply:
Could you point me to an example of a module which does this well?

Reply:
> Could you point me to an example of a module which does this well?

In terms of settings? Macros is a decent example.

Settings is pretty simple.

(I forgot the exact names of functions as I'm on mobile.)

In the first function you give it settings to store as an array.
In the second it gives you the array you last gave it.

If you define a variable top level (at the start) and then give it to the function which gets the settings, it will save it.

Then when loading it, in the arguments load it as something named similar and then set the variable you want to it.

JS example (not using object structure but generic): ```js
let thingToSave = 'thing!';

const getSettings = () => [ thingToSave ];
const loadSettings = ([ _thingToSave ]) => {
  thingToSave = _thingToSave;
};
```

Reply:
and this will save the settings too?

Reply:
> and this will save the settings too?

That code isn't direct but that's how the two handler functions act and how most modules use it.

Reply:
alright then.